src/Std/Data/DHashMap/Internal/List/Associative.lean

```diff
@@ -16,6 +17,22 @@ the contents of this file.
 File contents: Verification of associative lists
 -/
 
+
+-- TODO: should this go somwhere else?
+theorem List.find?_isSome_of_map_fst_contains {α : Type u} {β : α -> Type v} [BEq α] {l : List ((a : α) × β a)} {k : α} :
+    (l.map Sigma.fst).contains k ->
+      ((l.map Sigma.fst).reverse.find? (fun a => k == a)).isSome := by
+  intro h
+  simp
```

Comment:
All non-terminal `simp` or `simp at` calls must be converted into `simp only` calls via `simp?`. Otherwise the proofs are too fragile against changes to the simp set.

Reply:
Should be addressed (if we did not miss anything).